nilearn/datasets/struct.py

```diff
@@ -459,7 +510,16 @@ def fetch_surf_fsaverage5(data_dir=None, url=None, resume=True, verbose=1):
     ----------
     Fischl et al, (1999). High-resolution intersubject averaging and a
     coordinate system for the cortical surface. Hum Brain Mapp 8, 272-284.
+
     """
+    warnings.warn("fetch_surf_fsaverage5 has been deprecated and will "
+                  "be removed in a future release. "
+                  "Use fetch_surf_fsaverage(mesh='fsaverage5')",
+                  DeprecationWarning)
+    return fetch_surf_fsaverage(mesh='fsaverage5', data_dir=data_dir)
+
+
+def _fetch_surf_fsaverage5(data_dir=None, url=None, resume=True, verbose=1):
```

Comment:
Make `data_dir` a required arg, not optional. Same reasons as above. 
Since `_fetch_surf_fsaverage5()` is private, making `data_dir` required arg seems better to me.

Reply:
Here, `_fetch_surf_fsaverage5()` will only be used/called by `fetch_surf_fsaverage()`, not a user, so a default value of `data_dir=None` opens up this possibility:

    def _fetch_surf_fsaverage5(data_dir=None):
        print(data_dir)

    def fetch_surf_fsaverage(data_dir=None):
        _fetch_surf_fsaverage5()

    fetch_surf_fs_average(data_dir='some dir path')
    
    output>> None

Reply:
thanks!